webapp/src/scriptmanager.tsx

```diff
@@ -274,6 +293,133 @@ export class ScriptManagerDialog extends data.Component<ScriptManagerDialogProps
         this.setState({ sortedAsc: !sortedAsc, markedNew: {}, selected: {} });
     }
 
+    handleDownloadAsync = async () => {
+        pxt.tickEvent("scriptmanager.downloadZip", undefined, { interactiveConsent: true });
+
+        await loadZipAsync();
+
+        const { selected } = this.state;
+        const zipWriter = new zip.ZipWriter(new zip.Data64URIWriter("application/zip"));
+
+        const allHeaders = this.getSortedHeaders();
+        const selectedHeaders: pxt.workspace.Header[] = [];
+        Object.keys(selected).forEach((selectedIndex) =>  {
```

Comment:
Calling attention to this section, where I had to change the lookup logic since we don't have id's in this branch.